In [1]:
!pip install selenium

You should consider upgrading via the 'c:\users\91890\anaconda3\python.exe -m pip install --upgrade pip' command.


In [2]:
import pandas as pd
import selenium
from selenium import webdriver
import time
from selenium.common.exceptions import StaleElementReferenceException, NoSuchElementException

In [3]:
driver=webdriver.Chrome(r"C:\chromedriver.exe")

# from olx

In [90]:
url='https://www.olx.in/items/q-cars?isSearchCall=true'
driver.get(url)

In [91]:
#_89yzn

Price=[]
YearandDistancedone=[]
BrandandGas=[]

In [ ]:
for i in range(400):
    next_button = driver.find_element_by_xpath('//li[@class="_20FqS"]//div//button')      
    next_button.click()    

In [108]:



price=driver.find_elements_by_xpath("//span[@class='_89yzn']")
for i in price:
    try:
        Price.append(i.text)
    except NoSuchElementException:
        Price.append("--")

yearandDistancedone=driver.find_elements_by_xpath("//span[@class='_2TVI3']")
for i in yearandDistancedone:
    try:
        YearandDistancedone.append(i.text)
    except NoSuchElementException:
        YearandDistancedone.append("--")

brand=driver.find_elements_by_xpath("//span[@class='_2tW1I']")
for i in brand:
    try:
        BrandandGas.append(i.text)
    except NoSuchElementException:
        BrandandGas.append("--")

       

In [109]:
len(Price)

967

In [133]:
len(YearandDistancedone)

887

In [134]:
len(BrandandGas)

887

In [135]:
A=len(Price)-len(YearandDistancedone)

for i in range(A):
          Price.pop()
        
B=len(BrandandGas)-len(YearandDistancedone)

for i in range(B):
           BrandandGas.pop()

In [136]:
len(Price)

887

In [137]:
#creating dataframe
df=pd.DataFrame({'Price':Price,
                'BrandandGas':BrandandGas,
                'yearandDistancedone':YearandDistancedone})
df

Price                                        BrandandGas  \
0     ₹ 2,25,000                   Hyundai i10 Sportz, 2011, Petrol   
1     ₹ 8,20,000      Maruti Suzuki Vitara Brezza LDi, 2018, Diesel   
2    ₹ 10,70,000  Toyota Innova 2.5 GX (Diesel) 7 Seater, 2015, ...   
3     ₹ 3,20,000  Maruti Suzuki Wagon R Stingray 2014 CNG & Hybrids   
4     ₹ 2,50,000                            Tata Safari 2011 Diesel   
..           ...                                                ...   
882   ₹ 4,70,000        Maruti Suzuki Swift Dzire VDI, 2008, Diesel   
883   ₹ 4,45,000     Maruti Suzuki Alto 2007 Petrol 78523 Km Driven   
884   ₹ 1,55,000                   Hyundai EON Sportz, 2015, Petrol   
885   ₹ 4,90,000  Maruti Suzuki Alto 800 2010 Petrol Good Condition   
886   ₹ 6,65,000  Skoda Rapid 2013-2016 1.5 TDI Ambition With Al...   

    yearandDistancedone  
0      2011 - 60,000 km  
1      2018 - 17,000 km  
2     2015 - 145,000 km  
3      2014 - 50,000 km  
4     2011 - 160,000 km  
..                  ...  
882    2011 - 67,000 km  
883    2010 - 99,000 km  
884   2007 - 104,000 km  
885    2014 - 49,000 km  
886    2008 - 81,000 km  

[887 rows x 3 columns]

In [138]:
#splitting yearandDistancedone
df[['year2','distance']] = df["yearandDistancedone"].str.split("-",expand=True)
df.drop(columns=["yearandDistancedone"],inplace=True)
df

Price                                        BrandandGas  year2  \
0     ₹ 2,25,000                   Hyundai i10 Sportz, 2011, Petrol  2011    
1     ₹ 8,20,000      Maruti Suzuki Vitara Brezza LDi, 2018, Diesel  2018    
2    ₹ 10,70,000  Toyota Innova 2.5 GX (Diesel) 7 Seater, 2015, ...  2015    
3     ₹ 3,20,000  Maruti Suzuki Wagon R Stingray 2014 CNG & Hybrids  2014    
4     ₹ 2,50,000                            Tata Safari 2011 Diesel  2011    
..           ...                                                ...    ...   
882   ₹ 4,70,000        Maruti Suzuki Swift Dzire VDI, 2008, Diesel  2011    
883   ₹ 4,45,000     Maruti Suzuki Alto 2007 Petrol 78523 Km Driven  2010    
884   ₹ 1,55,000                   Hyundai EON Sportz, 2015, Petrol  2007    
885   ₹ 4,90,000  Maruti Suzuki Alto 800 2010 Petrol Good Condition  2014    
886   ₹ 6,65,000  Skoda Rapid 2013-2016 1.5 TDI Ambition With Al...  2008    

        distance  
0      60,000 km  
1      17,000 km  
2     145,000 km  
3      50,000 km  
4     160,000 km  
..           ...  
882    67,000 km  
883    99,000 km  
884   104,000 km  
885    49,000 km  
886    81,000 km  

[887 rows x 4 columns]

In [139]:
import re

Brandsplit=[]
for i in range(887):
        s = df["BrandandGas"][i]
        res = re.split('(\d+)', s)
        Brandsplit.append(res)
        
dataframe=pd.DataFrame(Brandsplit, columns=['1','2','3','4','5','6','7','8','9','10','11','12','13','14','15'])

In [140]:
dataframe

1     2               3      4  \
0                            Hyundai i    10        Sportz,    2011   
1    Maruti Suzuki Vitara Brezza LDi,   2018        , Diesel   None   
2                       Toyota Innova      2               .      5   
3      Maruti Suzuki Wagon R Stingray   2014   CNG & Hybrids   None   
4                         Tata Safari   2011          Diesel   None   
..                                 ...   ...             ...    ...   
882    Maruti Suzuki Swift Dzire VDI,   2008        , Diesel   None   
883                Maruti Suzuki Alto   2007         Petrol   78523   
884               Hyundai EON Sportz,   2015        , Petrol   None   
885                Maruti Suzuki Alto    800                   2010   
886                       Skoda Rapid   2013               -   2016   

                          5     6          7     8  \
0                  , Petrol  None       None  None   
1                      None  None       None  None   
2              GX (Diesel)      7   Seater,   2015   
3                      None  None       None  None   
4                      None  None       None  None   
..                      ...   ...        ...   ...   
882                    None  None       None  None   
883               Km Driven  None       None  None   
884                    None  None       None  None   
885   Petrol Good Condition  None       None  None   
886                             1          .     5   

                                    9    10         11    12    13    14    15  
0                                None  None       None  None  None  None  None  
1                                None  None       None  None  None  None  None  
2                            , Diesel  None       None  None  None  None  None  
3                                None  None       None  None  None  None  None  
4                                None  None       None  None  None  None  None  
..                                ...   ...        ...   ...   ...   ...   ...  
882                              None  None       None  None  None  None  None  
883                              None  None       None  None  None  None  None  
884                              None  None       None  None  None  None  None  
885                              None  None       None  None  None  None  None  
886   TDI Ambition With Alloy Wheel,   2014  , Diese..  None  None  None  None  

[887 rows x 15 columns]

In [141]:
import re

Brandsplit=[]
for i in range(887):
        s = dataframe["1"][i]
        res = re.findall('(\d+|[A-Za-z]+)', s)
        Brandsplit.append(res)

dataframe2=pd.DataFrame(Brandsplit, columns=['1','2','3','4','5','6','7','8','9','10'])

In [142]:
dataframe2

1       2       3       4         5     6     7     8     9    10
0    Hyundai       i    None    None      None  None  None  None  None  None
1     Maruti  Suzuki  Vitara  Brezza       LDi  None  None  None  None  None
2     Toyota  Innova    None    None      None  None  None  None  None  None
3     Maruti  Suzuki   Wagon       R  Stingray  None  None  None  None  None
4       Tata  Safari    None    None      None  None  None  None  None  None
..       ...     ...     ...     ...       ...   ...   ...   ...   ...   ...
882   Maruti  Suzuki   Swift   Dzire       VDI  None  None  None  None  None
883   Maruti  Suzuki    Alto    None      None  None  None  None  None  None
884  Hyundai     EON  Sportz    None      None  None  None  None  None  None
885   Maruti  Suzuki    Alto    None      None  None  None  None  None  None
886    Skoda   Rapid    None    None      None  None  None  None  None  None

[887 rows x 10 columns]

In [145]:
df["Brand"]=dataframe2["1"]
df["Model"]=dataframe2["2"]
df.drop(columns=["BrandandGas"],inplace=True)

In [146]:
df

Price  year2     distance    Brand   Model
0     ₹ 2,25,000  2011     60,000 km  Hyundai       i
1     ₹ 8,20,000  2018     17,000 km   Maruti  Suzuki
2    ₹ 10,70,000  2015    145,000 km   Toyota  Innova
3     ₹ 3,20,000  2014     50,000 km   Maruti  Suzuki
4     ₹ 2,50,000  2011    160,000 km     Tata  Safari
..           ...    ...          ...      ...     ...
882   ₹ 4,70,000  2011     67,000 km   Maruti  Suzuki
883   ₹ 4,45,000  2010     99,000 km   Maruti  Suzuki
884   ₹ 1,55,000  2007    104,000 km  Hyundai     EON
885   ₹ 4,90,000  2014     49,000 km   Maruti  Suzuki
886   ₹ 6,65,000  2008     81,000 km    Skoda   Rapid

[887 rows x 5 columns]

In [147]:
#removing ₹ from Price
df['Price'] = df['Price'].str.replace(r'₹', '')
df

Price  year2     distance    Brand   Model
0      2,25,000  2011     60,000 km  Hyundai       i
1      8,20,000  2018     17,000 km   Maruti  Suzuki
2     10,70,000  2015    145,000 km   Toyota  Innova
3      3,20,000  2014     50,000 km   Maruti  Suzuki
4      2,50,000  2011    160,000 km     Tata  Safari
..          ...    ...          ...      ...     ...
882    4,70,000  2011     67,000 km   Maruti  Suzuki
883    4,45,000  2010     99,000 km   Maruti  Suzuki
884    1,55,000  2007    104,000 km  Hyundai     EON
885    4,90,000  2014     49,000 km   Maruti  Suzuki
886    6,65,000  2008     81,000 km    Skoda   Rapid

[887 rows x 5 columns]

In [148]:
#removing km
#removing ₹ from Price
df['distance'] = df['distance'].str.replace(r'km', '')
df

Price  year2   distance    Brand   Model
0      2,25,000  2011     60,000   Hyundai       i
1      8,20,000  2018     17,000    Maruti  Suzuki
2     10,70,000  2015    145,000    Toyota  Innova
3      3,20,000  2014     50,000    Maruti  Suzuki
4      2,50,000  2011    160,000      Tata  Safari
..          ...    ...        ...      ...     ...
882    4,70,000  2011     67,000    Maruti  Suzuki
883    4,45,000  2010     99,000    Maruti  Suzuki
884    1,55,000  2007    104,000   Hyundai     EON
885    4,90,000  2014     49,000    Maruti  Suzuki
886    6,65,000  2008     81,000     Skoda   Rapid

[887 rows x 5 columns]

In [149]:
df.to_excel("Car prices.xlsx", index = False)

# from cardekho

In [4]:
url='https://www.cardekho.com/used-cars+in+bangalore'
driver.get(url)

In [7]:
Price=[]
Brand=[]
distancegasengine=[]
distance=[]
gas=[]
engine=[]



In [6]:
 for i in range(700):
            driver.execute_script("window.scrollBy(0, 250)")
            time.sleep(3)



In [8]:
Distancedone=driver.find_elements_by_xpath("//div[@class='truncate dotlist']//span")
for i in Distancedone:
    try:
        distancegasengine.append(i.text)
    except NoSuchElementException:
        distancegasengine.append("--")
        
for i in range(0,len(distancegasengine),3):
    distance.append(distancegasengine[i])

for i in range(1,len(distancegasengine),3):
    gas.append(distancegasengine[i])
    
for i in range(2,len(distancegasengine),3):
    engine.append(distancegasengine[i])

In [12]:
#finding product URLs

product_urls = []
url=driver.find_elements_by_xpath('//div[@class="gsc_col-xs-7 carsName"]//a')       #scraping urls
for i in url:
    product_urls.append(i.get_attribute("href"))  
    

In [15]:


for url in product_urls:
    driver.get(url)
    time.sleep(2)

    #extracting brand name and year
    brand=driver.find_elements_by_xpath("//div[@class='gsc_col-xs-12']//h1")
    for i in brand:
        try:
            Brand.append(i.text)
        except NoSuchElementException:
            Brand.append("--")

    
    #Modelname
    ModelName=[]
    modelName=driver.find_elements_by_xpath("//div[@class='listIcons']//div")
    for i in modelName:
        try:
            ModelName.append(i.text)
        except NoSuchElementException:
            ModelName.append("--")

    #emi
    EMI=[]
    emi=driver.find_elements_by_xpath("//div[@class='emitextF']//span")
    for i in emi:
        try:
            EMI.append(i.text)
        except NoSuchElementException:
            EMI.append("--")

            

BrandName=[]
year=[]

#extracting brand name        
for i in range(0,len(brand),2):
    BrandName.append(brand[i])


#extracting year        
for i in range(0,len(brand),2):
    year.append(brand[i])

In [26]:
Price=[]
for url in product_urls:
    driver.get(url)
    time.sleep(1)
    
    #extracting ruppee symbol and Price
    Price=[]
    price=driver.find_elements_by_xpath("//span[@class='pricewrapper posR']")
    for i in price:
        try:
            Price.append(i.text)
        except NoSuchElementException:
            Price.append("--")

In [30]:
Price

['Rs.2.85 Lakh*']

In [31]:
price

[<selenium.webdriver.remote.webelement.WebElement (session="4c97c914c6a0e2fd0e760bd86c698c5a", element="32250868-0613-4e70-bbbe-0e0186c43c29")>]

In [35]:
print("distance:",len(distance))
print("gas:",len(gas))
print("engine:",len(engine))
print("Price:",len(Price))
print("BrandName:",len(BrandName))
print("year:",len(year))

distance: 916
gas: 916
engine: 916
Price: 1
BrandName: 1
year: 1


In [37]:
#creating dataframe
df=pd.DataFrame({'engine':engine,
                'distance':distance,
                'gas':gas})
df

engine    distance     gas
0       Manual  44,479 kms  Petrol
1       Manual   9,418 kms  Petrol
2       Manual  30,045 kms  Petrol
3    Automatic  36,888 kms  Petrol
4       Manual  55,628 kms  Petrol
..         ...         ...     ...
911     Manual  27,383 kms  Diesel
912     Manual  10,887 kms  Diesel
913     Manual  89,209 kms  Diesel
914  Automatic  69,285 kms  Diesel
915     Manual  69,141 kms  Petrol

[916 rows x 3 columns]

In [41]:
df[['year', 'brand','model']] = df['Brand'].str.split(' ', 2, expand=True)
df.drop(columns=["Brand"],inplace=True)

In [42]:
df

Price    distance     gas        model  year    brand
0     5.55 Lakh  58,703 kms  Petrol        Swift  2014   Maruti
1      ₹ 11,238  44,479 kms  Petrol        Swift  2018   Maruti
2     7.14 Lakh  65,395 kms  Diesel          i20  2015  Hyundai
3      ₹ 14,447   7,751 kms  Petrol     Alto K10  2018   Maruti
4     6.72 Lakh   9,916 kms  Petrol  Swift Dzire  2016   Maruti
...         ...         ...     ...          ...   ...      ...
5287         --  83,000 kms  Diesel         None    --     None
5288         --  70,000 kms  Diesel         None    --     None
5289         --  50,000 kms  Diesel         None    --     None
5290         --  50,000 kms  Diesel         None    --     None
5291         --  70,000 kms  Petrol         None    --     None

[5292 rows x 6 columns]

In [43]:
df.to_excel("Car prices2.xlsx", index = False)

# from cars24

In [4]:
url='https://www.cars24.com/buy-used-cars-new-delhi/?itm_source=Cars24Website&itm_medium=sticky_header'
driver.get(url)

In [5]:
Price=[]
Brand=[]
distancegas=[]
distance=[]
gas=[]
Model=[]


In [6]:
for i in range(500):
            driver.execute_script("window.scrollBy(0, 250)")
            time.sleep(3)

In [8]:
price=driver.find_elements_by_xpath("//div[@class='_7udZZ']//span")
for i in price:
    try:
        Price.append(i.text)
    except NoSuchElementException:
        Price.append("--")
        

In [7]:



brand=driver.find_elements_by_xpath("//h2[@class='_3FpCg']")
for i in brand:
    try:
        Brand.append(i.text)
    except NoSuchElementException:
        Brand.append("--")
        

Distancegas=driver.find_elements_by_xpath("//ul[@class='bVR0c']")
for i in Distancegas:
    try:
        distancegas.append(i.text)
    except NoSuchElementException:
        distancegas.append("--")


In [72]:
#creating dataframe
df=pd.DataFrame({'Price':Price,
                'Brand':Brand,
                'distancegas':distancegas})
df

Price                    Brand                         distancegas
0     ₹4,11,699        2016 Maruti Swift        29,893 km\n1st Owner\nPetrol
1     ₹2,90,699     2014 Maruti Alto 800        12,535 km\n1st Owner\nPetrol
2     ₹2,63,599     2017 Maruti Alto 800        18,113 km\n1st Owner\nPetrol
3     ₹3,87,599     2019 Maruti Alto K10        16,224 km\n1st Owner\nPetrol
4     ₹1,94,999           2016 Tata Nano        20,039 km\n1st Owner\nPetrol
...         ...                      ...                                 ...
2284  ₹4,67,999  2014 Maruti Swift Dzire      1,22,399 km\n1st Owner\nDiesel
2285  ₹3,10,099         2020 Maruti Alto        24,411 km\n1st Owner\nPetrol
2286  ₹2,93,099         2019 Maruti Alto        41,638 km\n1st Owner\nPetrol
2287  ₹4,55,799      2018 Maruti Celerio  30,920 km\n1st Owner\nPetrol + CNG
2288  ₹3,00,799        2011 Toyota Etios      1,20,451 km\n2nd Owner\nPetrol

[2289 rows x 3 columns]

In [73]:
import re

Brandsplit=[]
for i in range(2289):
        s = df["Brand"][i]
        res = re.split('(\d+)', s)
        Brandsplit.append(res)
        

In [74]:
dataframe=pd.DataFrame(Brandsplit, columns=['a','b','c','d','e','f','g'])


In [75]:
dataframe

a      b                    c     d     e     f     g
0        2016         Maruti Swift  None  None  None  None
1        2014         Maruti Alto    800        None  None
2        2017         Maruti Alto    800        None  None
3        2019        Maruti Alto K    10        None  None
4        2016            Tata Nano  None  None  None  None
...  ..   ...                  ...   ...   ...   ...   ...
2284     2014   Maruti Swift Dzire  None  None  None  None
2285     2020          Maruti Alto  None  None  None  None
2286     2019          Maruti Alto  None  None  None  None
2287     2018       Maruti Celerio  None  None  None  None
2288     2011         Toyota Etios  None  None  None  None

[2289 rows x 7 columns]

In [76]:
df['year']=dataframe['b']
df['BrandModel']=dataframe['c']  
df.drop(columns=["Brand"],inplace=True)
df

Price                         distancegas  year           BrandModel
0     ₹4,11,699        29,893 km\n1st Owner\nPetrol  2016         Maruti Swift
1     ₹2,90,699        12,535 km\n1st Owner\nPetrol  2014         Maruti Alto 
2     ₹2,63,599        18,113 km\n1st Owner\nPetrol  2017         Maruti Alto 
3     ₹3,87,599        16,224 km\n1st Owner\nPetrol  2019        Maruti Alto K
4     ₹1,94,999        20,039 km\n1st Owner\nPetrol  2016            Tata Nano
...         ...                                 ...   ...                  ...
2284  ₹4,67,999      1,22,399 km\n1st Owner\nDiesel  2014   Maruti Swift Dzire
2285  ₹3,10,099        24,411 km\n1st Owner\nPetrol  2020          Maruti Alto
2286  ₹2,93,099        41,638 km\n1st Owner\nPetrol  2019          Maruti Alto
2287  ₹4,55,799  30,920 km\n1st Owner\nPetrol + CNG  2018       Maruti Celerio
2288  ₹3,00,799      1,20,451 km\n2nd Owner\nPetrol  2011         Toyota Etios

[2289 rows x 4 columns]

In [77]:
#splitting the data
df[['distance', 'owner', 'Gas']] = df['distancegas'].str.split('\n', 2, expand=True)
df.drop(columns=["distancegas"],inplace=True)



In [78]:
df

Price  year           BrandModel     distance      owner  \
0     ₹4,11,699  2016         Maruti Swift    29,893 km  1st Owner   
1     ₹2,90,699  2014         Maruti Alto     12,535 km  1st Owner   
2     ₹2,63,599  2017         Maruti Alto     18,113 km  1st Owner   
3     ₹3,87,599  2019        Maruti Alto K    16,224 km  1st Owner   
4     ₹1,94,999  2016            Tata Nano    20,039 km  1st Owner   
...         ...   ...                  ...          ...        ...   
2284  ₹4,67,999  2014   Maruti Swift Dzire  1,22,399 km  1st Owner   
2285  ₹3,10,099  2020          Maruti Alto    24,411 km  1st Owner   
2286  ₹2,93,099  2019          Maruti Alto    41,638 km  1st Owner   
2287  ₹4,55,799  2018       Maruti Celerio    30,920 km  1st Owner   
2288  ₹3,00,799  2011         Toyota Etios  1,20,451 km  2nd Owner   

               Gas  
0           Petrol  
1           Petrol  
2           Petrol  
3           Petrol  
4           Petrol  
...            ...  
2284        Diesel  
2285        Petrol  
2286        Petrol  
2287  Petrol + CNG  
2288        Petrol  

[2289 rows x 6 columns]

In [79]:
df2 = df['BrandModel'].str.split(n=1, expand=True)
df2.columns = ['STATUS_ID{}'.format(x+1) for x in df2.columns]
df = df.join(df2)

#renaming statusid1 and statusid2
df.rename(columns={'STATUS_ID1': 'Brand',
                   'STATUS_ID2': 'Model'},
          inplace=True, errors='raise')

In [80]:
df

Price  year           BrandModel     distance      owner  \
0     ₹4,11,699  2016         Maruti Swift    29,893 km  1st Owner   
1     ₹2,90,699  2014         Maruti Alto     12,535 km  1st Owner   
2     ₹2,63,599  2017         Maruti Alto     18,113 km  1st Owner   
3     ₹3,87,599  2019        Maruti Alto K    16,224 km  1st Owner   
4     ₹1,94,999  2016            Tata Nano    20,039 km  1st Owner   
...         ...   ...                  ...          ...        ...   
2284  ₹4,67,999  2014   Maruti Swift Dzire  1,22,399 km  1st Owner   
2285  ₹3,10,099  2020          Maruti Alto    24,411 km  1st Owner   
2286  ₹2,93,099  2019          Maruti Alto    41,638 km  1st Owner   
2287  ₹4,55,799  2018       Maruti Celerio    30,920 km  1st Owner   
2288  ₹3,00,799  2011         Toyota Etios  1,20,451 km  2nd Owner   

               Gas   Brand        Model  
0           Petrol  Maruti        Swift  
1           Petrol  Maruti        Alto   
2           Petrol  Maruti        Alto   
3           Petrol  Maruti       Alto K  
4           Petrol    Tata         Nano  
...            ...     ...          ...  
2284        Diesel  Maruti  Swift Dzire  
2285        Petrol  Maruti         Alto  
2286        Petrol  Maruti         Alto  
2287  Petrol + CNG  Maruti      Celerio  
2288        Petrol  Toyota        Etios  

[2289 rows x 8 columns]

In [83]:
#removing ₹ from Price
df['Price'] = df['Price'].str.replace(r'₹', '')
df

Price  year           BrandModel     distance      owner  \
0     4,11,699  2016         Maruti Swift    29,893 km  1st Owner   
1     2,90,699  2014         Maruti Alto     12,535 km  1st Owner   
2     2,63,599  2017         Maruti Alto     18,113 km  1st Owner   
3     3,87,599  2019        Maruti Alto K    16,224 km  1st Owner   
4     1,94,999  2016            Tata Nano    20,039 km  1st Owner   
...        ...   ...                  ...          ...        ...   
2284  4,67,999  2014   Maruti Swift Dzire  1,22,399 km  1st Owner   
2285  3,10,099  2020          Maruti Alto    24,411 km  1st Owner   
2286  2,93,099  2019          Maruti Alto    41,638 km  1st Owner   
2287  4,55,799  2018       Maruti Celerio    30,920 km  1st Owner   
2288  3,00,799  2011         Toyota Etios  1,20,451 km  2nd Owner   

               Gas   Brand        Model  
0           Petrol  Maruti        Swift  
1           Petrol  Maruti        Alto   
2           Petrol  Maruti        Alto   
3           Petrol  Maruti       Alto K  
4           Petrol    Tata         Nano  
...            ...     ...          ...  
2284        Diesel  Maruti  Swift Dzire  
2285        Petrol  Maruti         Alto  
2286        Petrol  Maruti         Alto  
2287  Petrol + CNG  Maruti      Celerio  
2288        Petrol  Toyota        Etios  

[2289 rows x 8 columns]

In [84]:
len(Price)

2289

In [85]:
len(Brand)

2289

In [86]:
len(distancegas)

2289

In [87]:
df.to_excel("Car prices3.xlsx", index = False)

# Data mined

522 data is collected from OLX

5292 data from Cardekho

2289 data from cars24